apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/views/chat/ChatInfoView.kt

```diff
@@ -1308,6 +1343,77 @@ fun queueInfoText(info: Pair<RcvMsgInfo?, ServerQueueInfo>): String {
   return generalGetString(MR.strings.message_queue_info_server_info).format(json.encodeToString(qInfo), msgInfo)
 }
 
+fun setChatTTLAlert(
+  m: ChatModel,
+  rhId: Long?,
+  chatInfo: ChatInfo,
+  selectedChatTTL: MutableState<ChatItemTTL?>,
+  previousChatTTL: ChatItemTTL?,
+  progressIndicator: MutableState<Boolean>
+) {
+  val defaultTTL = chatModel.chatItemTTL.value
+  val previouslyUsedTTL = previousChatTTL ?: defaultTTL
+  val newTTLToUse = selectedChatTTL.value ?: defaultTTL
+
+  AlertManager.shared.showAlertDialog(
+    title = generalGetString(
+      if (newTTLToUse.neverExpires) {
+        MR.strings.disable_automatic_deletion_question
+      } else if (!previouslyUsedTTL.neverExpires || selectedChatTTL.value == null) {
+        MR.strings.change_automatic_deletion_question
+      } else MR.strings.enable_automatic_deletion_question),
+    text = generalGetString(if (newTTLToUse.neverExpires) MR.strings.disable_automatic_deletion_message else MR.strings.change_automatic_chat_deletion_message),
+    confirmText = generalGetString(if (newTTLToUse.neverExpires) MR.strings.disable_automatic_deletion else MR.strings.delete_messages),
+    onConfirm = { setChatTTL(m, rhId, chatInfo, selectedChatTTL, progressIndicator, previousChatTTL) },
+    onDismiss = { selectedChatTTL.value = previousChatTTL },
+    onDismissRequest = { selectedChatTTL.value = previousChatTTL },
+    destructive = true,
+  )
+}
+
+private fun setChatTTL(
+  m: ChatModel,
+  rhId: Long?,
+  chatInfo: ChatInfo,
+  chatTTL: MutableState<ChatItemTTL?>,
+  progressIndicator: MutableState<Boolean>,
+  previousChatTTL: ChatItemTTL?
+  ) {
+  progressIndicator.value = true
+  withBGApi {
+    try {
+      m.controller.setChatTTL(rhId, chatInfo.chatType, chatInfo.apiId, chatTTL.value)
+      afterSetChatTTL(m, chatInfo, progressIndicator)
+    } catch (e: Exception) {
+      chatTTL.value = previousChatTTL
+      afterSetChatTTL(m, chatInfo, progressIndicator)
+      AlertManager.shared.showAlertMsg(generalGetString(MR.strings.error_changing_message_deletion), e.stackTraceToString())
+    }
+  }
+}
+
+private fun afterSetChatTTL(m: ChatModel, chatInfo: ChatInfo, progressIndicator: MutableState<Boolean>) {
+  withApi {
+    try {
+      // this is using current remote host on purpose - if it changes during update, it will load correct chats
+      // redirectDisabled is set to true to prevent redirecting the current chat/chat items in case the chat changes while messages were updating
+      apiLoadMessages(
+        m.remoteHostId(),
+        chatInfo.chatType,
+        chatInfo.apiId,
+        contentTag = null,
+        pagination = ChatPagination.Initial(ChatPagination.INITIAL_COUNT),
+        replaceChat = true,
+        redirectDisabled = true
+      )
+    } catch (e: Exception) {
+      Log.e(TAG, "apiGetChat error: ${e.message}")
+    } finally {
+      progressIndicator.value = false
+    }
+  }
+}
+
```

Comment:
can this be replaced with `openChat` funciton from ChatListNavLinkView? What's so special in it so you have to introduce new call and parameters to the apiLoadMessages?

Reply:
In here there are 2 situations, this seemed to be the best way to handle it.

1. The api replies fast with deletion, in the case open chat would be fine and was my initial tough (not sure if it would also update the chat list preview tough)
2. API to delete is slow if you have many items, user can click in another chat, we still want to update the chat in chat list but we don't want to redirect, correct me if i'm wrong but in that cases this would randomly redirect you to the previous chat, that's not a good user experience, looks more like a bug from user perspective

Reply:
1. If you're unsure, check it:)
2. So before calling openChat, just make this check for chatId. Right before openChat. No need to put another params to the main loading function